src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/CommandLineParser.cs

```diff
@@ -25,13 +25,14 @@ internal class CommandSwitch
 
         internal CommandSwitch(string name, string abbreviation, SwitchType switchType)
         {
-            //ensure that either name doesn't start with '/' or '-'
+            //ensure that either name doesn't start with '--' or '-'
             //also convert to lower-case
-            if ((name[0] == '/') || (name[0] == '-'))
-                _name = (name.Substring(1)).ToLower(CultureInfo.InvariantCulture);
+            if (name.Length > 1 && name[0] == '-' && name[1] == '-')
+                _name = (name.Substring(2)).ToLower(CultureInfo.InvariantCulture);
             else
                 _name = name.ToLower(CultureInfo.InvariantCulture);
-            if ((abbreviation[0] == '/') || (abbreviation[0] == '-'))
+
+            if (abbreviation[0] == '-')
```

Comment:
Looks like this statement didn't handle the case if abbreviation starts with "--"